aips/passkeys.md

```diff
@@ -0,0 +1,504 @@
+
+### OIDB Accounts
+
+OIDB accounts present a unique and novel way to generate an on-chain account with an OpenID account. This is highly desirable as OIDB accounts are accessible across all devices with browsers. Additionally, most people have an account associated with one or more OIDC providers (e.g., Google) and are familiar with OAuth login flows. Lastly, recoverability is not limited to certain operating systems in the same way that passkeys are.
+
+On the other hand, there are tradeoffs to consider as well:
+
+1. There are centralization and liveness concerns both on the OIDC provider and the services that OIDB accounts require to function in a privacy preserving manner (e.g., pepper service, proving service) respectively.
+2. Unless the ephemeral secret key ($\mathsf{esk}$) associated with the OIDB account of the user is encrypted (i.e., with a password) or is a passkey itself, the plaintext ephemeral secret key is still available on the browser, leaving it potentially vulnerable to a malicious actor (but only during its short expiration window). Passkey private keys, on the other hand, are stored on the device when the passkey is backed up, often within a secure element on the device and not on the browser.
+
+## Specification
+
+### Terminology
+
+- [**OIDB account**](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-61.md): An OpenID blockchain (OIDB) account whose security and liveness is backed by an OIDC account (e.g., a Google account) rather than a secret key.
+- [**Relying party**](https://www.w3.org/TR/webauthn-3/#relying-party): The website, application, or service that the user is trying to access, which is responsible for verifying a signature from the user's passkey to ensure that the user is who they claim to be.
+- [**WebAuthn Authenticator**](https://www.w3.org/TR/webauthn-3/#authenticator): A cryptographic entity, existing in hardware or software, that can register a user with a given relying party and later assert possession of the registered public key credential, and optionally verify the user to the relying party.
+
+
+### Aptos Account Creation with Passkeys
+
+During the registration of a new WebAuthn credential, an asymmetric key (i.e., a **private key** and its associateed **public key**) is generated securely on that device. The public key associated with the WebAuthn credential can be used to derive an associated account on chain. The private key, which controls the account, will be secured by the device.
+
+The registration process is particularly important for several reasons:
+
+1. The public key associated with the credential is only revealed during registration. Assertion responses (signing with the passkey) do NOT return the public key. Thus, the registration response must be parsed and the relevant information about the credential should be stored properly and persisted in storage by the relying party.
```

Comment:
This will need to be addressed. Otherwise, it sounds like if you lose your **public** key, you lose your account, which is no bueno.

Reply:
(Good news is ECDSA supports [pubkey recovery](https://cryptobook.nakov.com/digital-signatures/ecdsa-sign-verify-examples): given a message and a signature, you can extract a set of up to 2 candidate pubkeys.)

Reply:
But this does mean that adopting passkeys for other signature schemes will **NOT** be a good idea unless this is addressed.

Reply:
This depends on the signature though, doesn't it?

Reply:
Feels like we really ought to have account creation and account rotation function that take in a vector of public keys so that the authentication key can be verified and each public key stored as an event...